prost-build/src/lib.rs

```diff
@@ -1168,6 +1168,31 @@ impl Config {
         let mut written = 0;
         entries.sort();
 
+        if depth == 0 {
```

Comment:
I think this is wrong place to solve this issue. I believe the issue stems from a mismatch in file names of in this function and the `compile_fds` function. In that function a mapping between module name and file name is created. `write_includes` should reuse that mapping.

https://github.com/tokio-rs/prost/blob/b3de938c21e9875f17e3a4c116203f9c2f8e8640/prost-build/src/lib.rs#L995-L1003

Reply:
well, compile_fds does not have to worry about the tree structure of modules.

write_includes does, as it recursively writes the module declarations with their corresponding `include`s.

The filenames of the modules were correct, the problem was that the logic itself would panic for the base case of the recursion.

For reference, this is the generated include file of my test:

```rust
// This file is @generated by prost-build.
include!("__.default.rs");
pub mod image {
    include!("image.rs");
}
pub mod post {
    pub mod content {
        include!("post.content.rs");
    }
    include!("post.rs");
}
pub mod user {
    include!("user.rs");
}
```

The `write_includes` function needs to write handle modules without a path differently.

I think I will just change the function itself rather that hack some fix into it

Reply:
Good job!